### PR TITLE
Use mono_repo for package:web

### PR DESCRIPTION
**Pull request by srujzs**
Splits package:web into two packages:
- package:web
- package:web_generator (not published)

The latter contains our tooling to generate the former, and used to be the tool directory.

- Refactors code to point to new paths (and using `Platform.script.resolve` so that the `bin/` files can be run from anywhere instead of within that directory)
- Moves directories like `third_party` and tests to their new appropriate locations
- Moves CI configurations to `mono_repo.yaml` and `mono_pkg.yaml` files and is now generated as `dart.yml` (used to be `build.yaml`)
- Updates READMEs to reflect the package split
- Adds new LICENSE, `pubspec.yaml`, and CHANGELOG files
- Removes `JSArrayExtension` which is no longer needed.

Refactors code to point to new paths and moves directories like third_party to their appropriate locations. Moves the CI configurations to mono_repo.yaml and mono_pkg.yaml files, and is now generated as dart.yml.

### CODE REVIEW

**Comment by kevmoo** [comment]
But...why? What does this get us?

**Comment by srujzs** [comment]
This came up with [analysis of an external PR](https://github.com/dart-lang/web/actions/runs/10513688196/job/29133568180?pr=286). Analysis with the latest dev version has started failing due to the generator using `JSArray.length` and `JSArray.[]`. Since those members didn't exist until recently in `dart:js_interop`, we were using the declarations in [`util.dart`](https://github.com/dart-lang/web/blob/01e861efc5547cbc7802d3262ec1a99037d4b57a/tool/generator/util.dart#L14). Once those members were added to `dart:js_interop`, it started using the definitions in the SDK instead implicitly because type members have higher precedence than extension members. However, when we added `@Since` annotations, analysis started to fail because the minimum SDK version of this package was not `>=3.6`.

The solutions were:
- Update the minimum SDK constraint. This may be okay, but it doesn't make sense to require a later SDK version even though none of the code users can actually use is impacted. This probably also isn't the last time we'll need to do this.
- Disable analysis for the generator (temporary, but not tractable long term).
- Use extension overrides for every single usage of those `JSArray` members (bad considering there are like ~25 usages).
- Separate the generation code from the interop definitions (this PR).

As a bonus, if we wanted, we could publish the generator as a separate package so users can use it to generate definitions that we don't emit in this package. This was discussed in https://github.com/dart-lang/web/issues/145 before, but is not really the primary goal of this PR.

**Review by kevmoo — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,76 +1 @@
...
-  emit code that is standards track and is not experimental to reduce the number
-  of breaking changes.
-
-## Generation and updating the package
-
-Most of the APIs in this package are generated from public assets. See
-[tool/README.md](https://github.com/dart-lang/web/tree/main/tool) for
-information on the spec and IDL versions the package was generated from, and for
-the process for updating the package.
+web/README.md
```
Comment (sigmundch) [inline README.md:1]: did you mean to add a link?

File: LICENSE [hunk 2]
```diff
@@ -1,4 +1,4 @@
-Copyright 2023, the Dart project authors.
+Copyright 2024, the Dart project authors.
```
Comment (sigmundch) [inline LICENSE:1]: nit: I think it's OK to keep this one as 2023

File: web_generator/third_party/mdn/LICENSE
Comment (sigmundch) [inline web_generator/third_party/mdn/LICENSE:1]: I think we should undo this part of the change - I believe that we are still supposed to keep `third_party` at the top level of the git repository.

Comment (srujzs) [reply web_generator/third_party/mdn/LICENSE:1]: Yup, it looks like it's supposed to be in the root directory. Done, thanks!

File: LICENSE (hunk 2 above)
Comment (srujzs) [reply LICENSE]: Done.

File: README.md (hunk 1 above)
Comment (srujzs) [reply README.md]: Yes :) Changed it to be a symlink.

**Review by sigmundch — APPROVED** [review]
Thanks!

**Pull request merged**